WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/feedback/ipp/GetIPPFeedbackSurveyUrl.kt

```
@@ -0,0 +1,73 @@
+package com.woocommerce.android.ui.payments.feedback.ipp
+
+import com.woocommerce.android.extensions.daysAgo
+import com.woocommerce.android.extensions.formatToYYYYmmDD
+import com.woocommerce.android.ui.payments.GetActivePaymentsPlugin
+import org.jetbrains.annotations.VisibleForTesting
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.WCInPersonPaymentsStore
+import java.util.Date
+import javax.inject.Inject
+
+class GetIPPFeedbackSurveyUrl @Inject constructor(
+    private val shouldShowFeedbackBanner: ShouldShowFeedbackBanner,
+    private val ippStore: WCInPersonPaymentsStore,
+    private val siteModel: SiteModel,
+    private val getActivePaymentsPlugin: GetActivePaymentsPlugin,
+) {
+    @Suppress("ReturnCount")
+    @Throws(IllegalStateException::class)
+    suspend operator fun invoke(): String? {
+        requireShouldShowFeedbackBanner()
+
+        val activePaymentsPlugin = requireWooCommercePaymentsPlugin()
+
+        val timeWindowStartDate = Date().daysAgo(STATS_TIME_WINDOW_LENGTH_DAYS)
+        val response = ippStore.fetchTransactionsSummary(
+            activePaymentsPlugin,
+            siteModel,
+            timeWindowStartDate.formatToYYYYmmDD()
+        )
+
+        if (response.isError || response.result == null) return null
```

❓ I think we might need the failure cause if we decide to log the failure event. In that case, we need to return the failure cause instead of null. WDYT?

---

Good idea!